### PR TITLE
Fix errors on section rendering caused by CORS issues

### PR DESCRIPTION
**Pull request by karreiro**
### WHY are these changes introduced?

The Dawn 1.1.0 theme relies on data attribute values to re-render sections with JavaScript:

Screenshot 2022-02-20 at 11 25 17

However, when users change product variants, the exact error reported by Shopify/shopify-cli#1466 happens, as the client-side request fails due to CORS issues.

#### Regarding my other ([reverted](https://github.com/Shopify/shopify-cli/pull/1985)) PR

My other PR (#1952) was actually fixing a different problem with less impact. It handled 307 HTTP requests with CORS errors happening only on Safari with some assets.

Those assets do not impact section rendering. So, I've created https://github.com/Shopify/cli/issues/2001 for handling this scenario.

### WHAT is this pull request doing?

This PR adopts a similar approach from the other middlewares and replaces shop URLs from templates, pointing them to `127.0.0.1:9292`.

### How to test your changes?

- Clone the Dawn theme `git clone git@github.com:Shopify/dawn.git`
- Go to version 1.1.0 with `git reset 2566749e4226818500809892259b7713808319ae --hard`
- Import products data from [products.csv](https://gist.githubusercontent.com/karreiro/2cc4724245a11e2549cc7e6711def453/raw/a11c91bc0cab186af910cbd5126de24102597b34/products_export.csv) in your store
- Run `shopify theme serve`
- Interact with the **Classic Varsity Top** and observe the browser console

**Before this PR:**
before

**After this PR:**
after

### Post-release steps

None.

### Update checklist

- [x] I've added a CHANGELOG entry for this PR (if the change is public-facing)
- [x] I've considered possible cross-platform impacts (Mac, Linux, Windows).
- [x] I've left the version number as is (we'll handle incrementing this when releasing).
- [x] I've included any post-release steps in the section above.

### CODE REVIEW

File: lib/shopify_cli/theme/dev_server/proxy.rb [hunk 1]
```diff
@@ -70,13 +69,23 @@ def call(env)
...
           body = [body] unless body.respond_to?(:each)
           [response.code, headers, body]
         end
 
         private
 
+        def patch_body(env, body)
+          return [""] unless body
+
+          body.gsub(%r{(http:|https:)?//#{@theme.shop}/}, "http://#{host(env)}/")
```
Comment (macournoyer) [inline lib/shopify_cli/theme/dev_server/proxy.rb:82]:
```suggestion
          body.gsub(%r{(https?:)?//#{@theme.shop}/}, "http://#{host(env)}/")
```

**Review by macournoyer — APPROVED** [review]
As this will replace ALL urls on the page, I wasn't sure if that was going to work with cart & checkout, but after some quick testing it seem to work (I did not test completing a full checkout flow, just submitting the form). 🙌

File: lib/shopify_cli/theme/dev_server/proxy.rb (hunk 1 above)
Comment (karreiro) [reply lib/shopify_cli/theme/dev_server/proxy.rb:82]: Nice improvement ✨Thanks, @macournoyer! :)

**Comment by karreiro** [comment]
Thanks for the review, @macournoyer! I've realized the regex could be more specific for solving the issue. So, I've changed it to handle only URLs that appear on data attributes.

In the future, if we feel we need to cover broader scenarios, we may make it more generic, but being specific seems safer now :)

Thanks again for the review! 🚀

**Pull request merged**